### PR TITLE
Candidate decider admin backend

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 
This PR completes the functionality of the admin-side page for Candidate Decider
- Added delete button and modal to confirm deleting instances 
- Implemented backend endpoints to support the admin page 
    - `getAllCandidateDeciderInstances`
    - `createNewCandidateDeciderInstance`
    - `toggleCandidateDeciderInstance` (toggles if an instance is open) 
    - `deleteCandidateDeciderInstance` 

### Test Plan 

Try creating, toggling, and deleting an instance on the deploy preview

### CODE REVIEW

File: frontend/src/API/CandidateDeciderAPI.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import APIWrapper from './APIWrapper';
+import { backendURL } from '../environment';
+
+export default class CandidateDeciderAPI {
+  static async getAllInstances(): Promise<CandidateDeciderInfo[]> {
+    const response = APIWrapper.get(`${backendURL}/getAllCandidateDeciderInstances`);
+    return response.then((val) => val.data.instances);
+  }
+
+  static async createNewInstance(instance: CandidateDeciderInstance) {
```
Comment (riyajaggi) [inline frontend/src/API/CandidateDeciderAPI.ts:10]: missing return type on function

File: backend/src/candidateDeciderAPI.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import CandidateDeciderDao from './dao/CandidateDeciderDao';
+import { PermissionError } from './errors';
+import PermissionsManager from './permissions';
+
+export const getAllCandidateDeciderInstances = async (
+  user: IdolMember
+): Promise<CandidateDeciderInfo[]> => {
+  const hasPermission = PermissionsManager.isAdmin(user);
```
[the remark this answers is not in the document]
Comment (henry-li-06) [reply backend/src/candidateDeciderAPI.ts]: I actually just removed this permission check, I think it's unnecessary for this endpoint handler. This handler returns `CandidateDeciderInfo` type which doesn't have any actually info about the candidates or their responses so it isn't much of a privacy issue for all members to see/know. We definitely need to include a more complex permission checking mechanism for the endpoint handler that actually fetches the response info.

[the remark this answers is not in the document]
Comment (henry-li-06) [reply backend/src/candidateDeciderAPI.ts]: Added `authorizedMembers` and `authorizedRoles` to `CandidateDeciderInstance` type. So to access the candidate info for an instance, the user either has to be lead or admin, have their role be authorized, or have the user themselves specifically authorized. And then for `getAllCandidateDeciderInstances`, there aren't any permission checks anymore.

[the remark this answers is not in the document]
Comment (riyajaggi) [reply backend/src/candidateDeciderAPI.ts]: This is really good! The option to add authorized members makes the candidate decider very flexible and adaptable to use.

**Comment by riyajaggi** [comment]
One more frontend thing: Whenever a new instance is created or deleted, the page doesn't refresh automatically to show the updated instances. We have to do it ourselves to see the new instance.

**Review by riyajaggi — APPROVED** [review]
This is great!

**Comment by henry-li-06** [comment]
> One more frontend thing: Whenever a new instance is created or deleted, the page doesn't refresh automatically to show the updated instances. We have to do it ourselves to see the new instance.

Should be fixed now

**Pull request merged**